src/pfs/server/api_server.go

```diff
@@ -164,11 +181,28 @@ func (a *apiServer) InspectCommit(ctx context.Context, request *pfs.InspectCommi
 	a.versionLock.RLock()
 	defer a.versionLock.RUnlock()
 	ctx = versionToContext(a.version, ctx)
-	clientConn, err := a.getClientConn(a.version)
+
+	clientConns, err := a.router.GetAllClientConns(a.version)
 	if err != nil {
 		return nil, err
 	}
-	return pfs.NewInternalAPIClient(clientConn).InspectCommit(ctx, request)
+
+	var commitInfos []*pfs.CommitInfo
+	for _, clientConn := range clientConns {
+		commitInfo, err := pfs.NewInternalAPIClient(clientConn).InspectCommit(ctx, request)
+		if err != nil {
+			return nil, err
+		}
+		commitInfos = append(commitInfos, commitInfo)
+	}
+
+	for _, commitInfo := range pfs.ReduceCommitInfos(commitInfos) {
+		if commitInfo.Commit.ID == request.Commit.ID {
+			 return commitInfo, nil
+		}
+	}
```

Comment:
Same as above we should make sure that this reduces down to a single CommitInfo, return it if it does, otherwise error.